### DISCUSSION TITLE
How to use libredwg on VisualStudio2022(windows x64)?

### DISCUSSION THREAD

**Discussion by BoTan-sudo** [LibreDWG/libredwg#798] [Q&A]
Hi, I want to use the libredwg in VisualStudio2022. I downloaded the libredwg-0.12.5.6027-win64.zip and extracted it. And the header file DIR and library file DIR are configured in visual studio 2022, here is my configure.
image1.
![image](https://github.com/LibreDWG/libredwg/assets/38968210/d5f20941-cce2-4b54-bc90-1fe3dd52aef5)
image2.
![image](https://github.com/LibreDWG/libredwg/assets/38968210/e31590a3-1dcd-4cec-8856-057d0e2ea3c3)
image3.
![image](https://github.com/LibreDWG/libredwg/assets/38968210/dc071960-85b2-4793-8c57-49943b8ddda7)

and here is my helloworld code:
```
/* Helloworld.hpp */
#pragma once

class HelloWorld {
public:
	void SayHello(void);
};
```

```
/* Helloworld.cpp */
#include "Helloworld.hpp"
#include <iostream>

void HelloWorld::SayHello(void)
{
	std::cout << "Hello World!\r\n" << std::endl;
}
```

```
/* main.cpp */

#include "Helloworld.hpp"

int main()
{	
	HelloWorld helloObj;
	helloObj.SayHello();
	return 0;
}
```

when I click the "rebuild" , the VisualStudio2022 report Error " LNK1107 invalid or corrupt file: cannot read at 0x4A8 in libredwg-0.dll "
How do I get this? Please help and guide on how to use the libredwg on VisualStudio2022.

**Comment by icfsoft** [comment]
You cannot use the pre built releases with VS2019/2022, you have to build it yourself from source. I just did it and it works.

**Reply by demoonj** [reply]
<img width="356" height="242" alt="ddfsafsdaf" src="https://github.com/user-attachments/assets/647a9015-9643-43ff-ad73-71b5daf09603" />

**Reply by demoonj** [reply]
Have you ever encountered such a problem when using it?
